setuptools/_importlib.py

```diff
@@ -12,6 +12,17 @@ def disable_importlib_metadata_finder(metadata):
         import importlib_metadata
     except ImportError:
         return
+    except AttributeError:
+        import warnings
+
+        msg = (
+            "`importlib_metadata` version is incompatible with `setuptools`.\n"
+            "This problem is likely to be solved by installing an updated version of "
+            "`importlib_metadata`."
```

Comment:
I think the canonical package name is `importlib-metadata`, since that's what's on PyPI:
https://pypi.org/project/importlib-metadata/

Since end users will be needing to look for the requirement, rather than using the import - would it be best if the warning message used `importlib-metadata` instead?

Reply:
Thank you very much for the review @edmorley.

I agree with you about `importlib-metadata`.